src/plugins/analysis/hardware_analysis/code/hardware_analysis.py

```diff
@@ -0,0 +1,88 @@
+from typing import Optional
+
+from analysis.PluginBase import AnalysisBasePlugin
+
+
+class AnalysisPlugin(AnalysisBasePlugin):
+    '''
+    Hardware Analysis Plug-in
+    '''
+    NAME = 'hardware_analysis'
+    DESCRIPTION = 'Hardware Analysis Plug-in'
+    DEPENDENCIES = ['cpu_architecture', 'elf_analysis', 'kernel_config']
+    VERSION = '0.2'
+
+    def __init__(self, plugin_adminstrator, config=None, recursive=True):
+
+        self.config = config
+
+        super().__init__(plugin_adminstrator, config=config, recursive=recursive, plugin_path=__file__)
+
+    def process_object(self, file_object):
+
+        # search for important information
+        cpu_architecture = self.cpu_architecture_analysis(file_object)
+        modinfo = self.filter_modinfo(file_object)
+        kernel_config = self.filter_kernel_config(file_object)
+
+        # store the results
+        file_object.processed_analysis[self.NAME] = dict()
+        file_object.processed_analysis[self.NAME]['cpu architecture'] = cpu_architecture
+        file_object.processed_analysis[self.NAME]['*.ko ".modinfo"'] = modinfo
+        file_object.processed_analysis[self.NAME]['kernel configuration'] = kernel_config
+
+        # propagate summary to parent objects
+        file_object.processed_analysis[self.NAME]['summary'] = self.make_summary(cpu_architecture, modinfo, kernel_config)
+
+        return file_object
+
+    def cpu_architecture_analysis(self, file_object) -> Optional[str]:
+        cpu_architecture_dict = file_object.processed_analysis['cpu_architecture']
+        cpu_architecture = cpu_architecture_dict["summary"]
+
+        if cpu_architecture == []:
+            cpu_architecture = None
+        else:
+            cpu_architecture = cpu_architecture[0]
+
+        return cpu_architecture
+
+    def filter_modinfo(self, file_object):
+        # getting the information from the *.ko files .modinfo
+        if "modinfo" in file_object.processed_analysis['elf_analysis'].keys():
+            modinfo = file_object.processed_analysis['elf_analysis']['modinfo']
+        else:
+            modinfo = None
+
+        return modinfo
+
+    def filter_kernel_config(self, file_object):
```

Comment:
We should filter out kernel config entries that are irrelevant (from hardware perspective). Displaying all entries is not very useful. This could be accomplished either by filtering out entries that are known to be irrelevant (aka "blacklisting") or keeping entries that are known to be useful (aka "whitelisting"). The former is prone to false positives and the latter is prone to false negatives. I guess the former makes more sense in this case, because we can't really know all relevant entries but there are a lot of default entries, that are not very informative.